lib/http.c

```diff
@@ -414,8 +415,41 @@ mapcache_http* mapcache_http_configuration_parse_xml(mapcache_context *ctx, ezxm
       apr_table_set(req->headers, header_node->name, header_node->txt);
     }
   }
+
+  /* Parse auth and append to headers for simplicity */
+  if ((http_node = ezxml_child(node, "auth")) != NULL) {
+    if (ezxml_attr(http_node, "scheme") &&
+        strcmp(ezxml_attr(http_node, "scheme"), "basic") == 0) {
+      ezxml_t user_node, pass_node;
+      const char *credentials;
+      user_node = ezxml_child(http_node, "user");
+      pass_node = ezxml_child(http_node, "pass");
+      if (!user_node || !pass_node) {
+        ctx->set_error(ctx, 400,
+                       "both <http> <auth> username (<user>) and pasword "
+                       "(<pass>) must be provided");
+        return NULL;
```

Comment:
How would you provide an empty password? We at the National Land Survey of Finland have some services that use api-key as `<user>` but `<password>` is not required. At least some clients can use those services with syntax `user:` (username-colon-nothing).

Reply:
I had not met such services yet. I'll adjust the code.

Reply:
I checked the code. No changes are required as only the presence of xml element is performed. Thus this just will have to be documented that user or pass elements are encoded verbatim and can be left empty (`<pass></pass>`) if not required.